src/test/munit/error-functions-test-suite.xml

```diff
@@ -78,7 +78,7 @@ http://www.mulesoft.org/schema/mule/java http://www.mulesoft.org/schema/mule/jav
         <munit:validation>
             <munit-tools:assert-equals actual="#[Java::isInstanceOf(payload, 'java.lang.NullPointerException')]"
                                        expected="#[true]"/>
-            <munit-tools:assert-equals actual="#[payload.detailMessage]" expected="#[vars.errorMessage]"/>
```

Comment:
Why is the property from which the error message is obtained changed? Couldn't the change in the assert break backwards compatibility?

Reply:
Issue has been discussed under W-14380221, brief desription: 
"this expression seems to be accessing things beyond the Mule error API. With Java 17, DataWeave can no longer access those sort of fields because the mechanism to do so was reflections. DataWeave will be able to access fields in Java classes that have the proper getters/setters: POJOs, essentially."

Reply:
Thanks @martina-tutokiova , it was clear to me that the change is necessary and would not affect backward compatibility.